custom_metrics/css.js

```diff
@@ -1,30 +1,62 @@
 //[css]
 // Uncomment the previous line for testing on webpagetest.org
 
+const PREFERS_COLOR_SCHEME_REGEXP =
+  /@media\s+\(\s*prefers-color-scheme\s*:\s*(?:dark|light)\s*\)\s*\{[^\}]*\}/gms;
+
+const bodies = $WPT_BODIES;
+
 return JSON.stringify({
-  'css_in_js': (() => {
+  css_in_js: (() => {
     const CssInJsMap = {
-      'Styled Components': !!document.querySelector('style[data-styled],style[data-styled-components]'),
-      'Radium': !!document.querySelector('[data-radium]'),
-      'JSS': !!document.querySelector('[data-jss]'),
-      'Emotion': !!document.querySelector('[data-emotion]'),
-      'Goober': !!document.getElementById('_goober'),
+      'Styled Components': !!document.querySelector(
+        'style[data-styled],style[data-styled-components]',
+      ),
+      Radium: !!document.querySelector('[data-radium]'),
+      JSS: !!document.querySelector('[data-jss]'),
+      Emotion: !!document.querySelector('[data-emotion]'),
+      Goober: !!document.getElementById('_goober'),
       'Merge Styles': !!document.querySelector('[data-merge-styles]'),
       'Styled Jsx': !!document.querySelector('style[id*="__jsx-"]'),
-      'Aphrodite': !!document.querySelector('[data-aphrodite]'),
-      'Fela': !!document.querySelector('[data-fela-stylesheet]'),
-      'Styletron': !!document.querySelector('[data-styletron],._styletron_hydrate_'),
-      'React Native for Web': !!document.querySelector('#react-native-stylesheet'),
-      'Glamor': !!document.querySelector('[data-glamor]')
-    }
+      Aphrodite: !!document.querySelector('[data-aphrodite]'),
+      Fela: !!document.querySelector('[data-fela-stylesheet]'),
+      Styletron: !!document.querySelector(
+        '[data-styletron],._styletron_hydrate_',
+      ),
+      'React Native for Web': !!document.querySelector(
+        '#react-native-stylesheet',
+      ),
+      Glamor: !!document.querySelector('[data-glamor]'),
+    };
 
-    const usedLibraries = []
+    const usedLibraries = [];
     for (l in CssInJsMap) {
       if (CssInJsMap[l]) {
         usedLibraries.push(l);
       }
     }
 
     return usedLibraries;
-  })()
+  })(),
+  // Checks in two passes:
+  // 1. The response body of same- and cross-origin stylesheets.
+  // 2. The `link[media]` attribute of conditionally loaded stylesheets in the
+  //    ternary expression if step 1. returns `false`.
+  prefersColorScheme:
+    // For all stylesheets…
+    bodies
+      .filter((request) => request.type === 'Stylesheet')
+      // …check if the `prefers-color-scheme` RegExp matches…
+      .map((request) =>
+        PREFERS_COLOR_SCHEME_REGEXP.test(request.response_body || ''),
+      )
+      // If even just one of the stylesheets matches, return `true`.
+      .filter((usesPrefersColorScheme) => usesPrefersColorScheme).length > 0 ||
```

Comment:
Could these three methods be combined into one?
```suggestion
    // …check if the `prefers-color-scheme` RegExp matches…
    // If even just one of the stylesheets matches, return `true`.
    bodies.some((request) => {
      return request.type === 'Stylesheet' && PREFERS_COLOR_SCHEME_REGEXP.test(request.response_body || '')
    }) ||
```